pkg/cmd/codespace/ports.go

```diff
@@ -380,3 +370,23 @@ func normalizeJSON(j []byte) []byte {
 	// remove trailing commas
 	return bytes.ReplaceAll(j, []byte("},}"), []byte("}}"))
 }
+
+func getCodespaceForPorts(ctx context.Context, apiClient apiClient, codespaceName string) (*api.Codespace, error) {
```

Comment:
Wondering if we shouldn't name this something more generic and use it everywhere else to DRY things up a bit?

Reply:
Yeah I think that's a good idea or just move the error handling into `getOrChooseCodespace` since I think we're now doing this error handling at every `getOrChooseCodespace` call site.

Reply:
Yeah that works!